src/OpenTelemetry.Api/Context/Propagation/BaggageFormat.cs

```diff
@@ -0,0 +1,146 @@
+﻿// <copyright file="BaggageFormat.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Context.Propagation
+{
+    /// <summary>
+    /// W3C baggage: https://github.com/w3c/baggage/blob/master/baggage/HTTP_HEADER_FORMAT.md.
+    /// </summary>
+    public class BaggageFormat : ITextFormat
+    {
+        private const string Baggage = "baggage";
+        private const int MaxBaggageLength = 1024;
+
+        /// <inheritdoc/>
+        public ISet<string> Fields => new HashSet<string> { Baggage };
+
+        /// <inheritdoc/>
+        public TextFormatContext Extract<T>(TextFormatContext context, T carrier, Func<T, string, IEnumerable<string>> getter)
+        {
+            if (carrier == null)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToInjectActivityContext("null carrier");
```

Comment:
typo - FailedToExtract.

Reply:
Fixed and improved the logging in all the ITextFormat implementations.